train.py

```diff
@@ -120,12 +120,18 @@ def weight_stats(self, iteration, model, log_weight_distribution, eps = 1e-9):
 				self.summary_writer.add_histogram(tag + '/grad', param.grad, iteration)
 	
 
-def apply_model(data_loader, model, generator, text_pipelines, device, oom_handler):
+def apply_model(data_loader, model, generator, text_pipelines, device, oom_handler, onnx_wrapper=None, forward_x_only=False):
 	for meta, s, x, xlen, y, ylen in data_loader:
 		x, xlen, y, ylen = x.to(device, non_blocking = True), xlen.to(device, non_blocking = True), y.to(device, non_blocking = True), ylen.to(device, non_blocking = True)
 		with torch.no_grad():
 			try:
-				logits, log_probs, olen, loss = map(model(x, xlen, y = y, ylen = ylen).get, ['logits', 'log_probs', 'olen', 'loss'])
+				model_to_infer = onnx_wrapper if onnx_wrapper is not None else model
+				infer_kwargs = {'x': x} if forward_x_only else {'x': x, 'xlen': xlen, 'y': y, 'ylen': ylen}
+				logits, log_probs, olen, loss = map(model_to_infer(**infer_kwargs).get, ['logits', 'log_probs', 'olen', 'loss'])
+
+				if loss is None:
```

Comment:
please move this hack into `OnnxWrapper`. please check what should be the correct `loss` tensor's shape and put a correctly-shaped dummy